src/Middleware/StaticFiles/test/UnitTests/DefaultFilesMiddlewareTests.cs

```diff
@@ -213,24 +227,34 @@ public async Task PostDirectory_PassesThrough_All(string baseUrl, string baseDir
         [InlineData("/SubFolder", @".\", "/SubFolder/")]
         [InlineData("", @".\SubFolder", "/")]
         [InlineData("", @".\SubFolder\", "/")]
-        public async Task PostDirectory_PassesThrough_Windows(string baseUrl, string baseDir, string requestUrl)
+        [InlineData("/SubFolder", @".\", "/SubFolder/", false)]
+        [InlineData("", @".\SubFolder", "/", false)]
+        [InlineData("", @".\SubFolder\", "/", false)]
+        public async Task PostDirectory_PassesThrough_Windows(string baseUrl, string baseDir, string requestUrl, bool appendTrailingSlash = true)
         {
-            await PostDirectory_PassesThrough(baseUrl, baseDir, requestUrl);
+            await PostDirectory_PassesThrough(baseUrl, baseDir, requestUrl, appendTrailingSlash);
         }
 
-        private async Task PostDirectory_PassesThrough(string baseUrl, string baseDir, string requestUrl)
+        private async Task PostDirectory_PassesThrough(string baseUrl, string baseDir, string requestUrl, bool appendTrailingSlash = true)
```

Comment:
All of the test scenarios you've added have the same result with and without the new option. There needs to be a test that confirms a redirect is not generated with the new option changed. See NearMatch_RedirectAddSlash.

Reply:
I've added new inline data without trailing slash at the end, if RedirectToAppendTrailingSlash wouldn't work those test would fail beacause of redirect, each test checks response.StatusCode.